### ISSUE TITLE
Allow multiple polls on the same page

### ISSUE THREAD

**Issue by lesterchan** [lesterchan/wp-polls#35]
Changes has to be made to not use id but class like .poll-1 and not #poll-id alongside with javascript changes that find the context of the poll button being clicked.

https://wordpress.org/support/topic/weird-things-happen-when-the-same-poll-is-showing-in-content-and-sidebar?replies=3#post-6512375

**Comment by espellcaste** [comment]
@lesterchan Any idea from the top of the head what changes would be needed, in case someone wishes to send a pull request?

**Comment by lesterchan** [comment]
@espellcaste lots of changing. First you need to modify the template to ensure that id="polls-{id}" is no longer being used, instead it should be class="polls-{id}" after that you need to modify the JS file to find the .polls-{id} being rated

**Comment by bitinn** [comment]
would love to see this, for years (I mean literally, been using this since 2006) i couldn't figure out a proper way to do this.

would it be easier to have a logic that hides the sidebar poll, when i manually added it to main content?

**Comment by espellcaste** [comment]
Got it @lesterchan! It seems not that complicated... I'm gonna try to suggest something in the future. =)

**Comment by bitinn** [comment]
I would suggest `data-id` instead of `class`.

**Comment by espellcaste** [comment]
Any advantages in this approach @bitinn?

**Comment by bitinn** [comment]
@espellcaste generally speaking, separating class name from javascript is a good idea, both for separation of concerns and ease of maintenance. In our case the poll id serve no actual styling purpose, but is used as selector in jquery, since repeat `id` is no-go, `data-*` attributes works best.

with `data-id` i expect not much needed to be changed, but I haven't looked into this plugin lately, so maybe @lesterchan can offer some suggestions.

**Comment by lesterchan** [comment]
I agree. I would go with data-poll_id=<ID> and .poll-<ID>. Some people want specified style for a poll, so i think have a specific class is good as well.

**Comment by espellcaste** [comment]
Good point @bitinn. I agree @lesterchan.

**Comment by archon810** [comment]
@lesterchan Any updates on this after 4 years? :P
